### PR TITLE
Sd/staticarrays cylinders without arrays compatible v0.6

### PR DESCRIPTION
**Pull request by edljk**

### CODE REVIEW

File: src/decompose.jl [hunk 1]
```diff
@@ -394,43 +394,43 @@ isdecomposable{T<:Point, C<:Cylinder2}(::Type{T}, ::Type{C}) = true
...
     for i = 1:length(vertices)
         vertices[i] = PT(M*(vertices[i]-vo)+vo)
     end
     return vertices
 end
-function decompose{T}(PT::Type{Point{3,T}}, c::Cylinder{3,T}, resolution = 5)
-    isodd(resolution) && (resolution = 2*div(resolution, 2))
-    resolution = max(8, resolution); nbv = div(resolution, 2)
+function decompose{T}(PT::Type{Point{3,T}},c::Cylinder{3,T},resolution=5)
+    isodd(resolution) ? resolution = 2*div(resolution,2) : nothing
```
Comment (SimonDanisch) [inline src/decompose.jl]: Did you revert this on purpose? I find `x ? a : nothing` is a pretty confusing pattern.

File: src/decompose.jl [hunk 2]
```diff
@@ -394,43 +394,43 @@ isdecomposable{T<:Point, C<:Cylinder2}(::Type{T}, ::Type{C}) = true
...
+    resolution<8 ? resolution = 8 : nothing; nbv = Int(resolution/2)
```
Comment (SimonDanisch) [inline src/decompose.jl]: same here, I think div is more appropriate in this situation

File: src/decompose.jl [hunk 3]
```diff
@@ -394,43 +394,43 @@ isdecomposable{T<:Point, C<:Cylinder2}(::Type{T}, ::Type{C}) = true
...
     M = rotation(c); h = height(c)
-    position = 1; vertices = Vector{PT}(2*nbv)
+    position = 1; vertices = Array{PT}(2*nbv)
     for j = 1:nbv
-        phi = T((2pi * (j - 1)) / nbv)
-        vertices[position] = PT(M * [c.r * cos(phi); c.r * sin(phi); 0]) + PT(c.origin)
-        vertices[position+1] = PT(M * [c.r * cos(phi); c.r * sin(phi); h]) + PT(c.origin)
+        phi = T((2*pi*(j-1))/nbv)
```
Comment (SimonDanisch) [inline src/decompose.jl]: You removed all my spaces I put in here... Would it be okay to put them back? I really like spaces after `,` and operators.
Also you turned `Vector{PT}` into `Array{T}`.

File: src/decompose.jl [hunk 4]
```diff
@@ -394,43 +394,43 @@ isdecomposable{T<:Point, C<:Cylinder2}(::Type{T}, ::Type{C}) = true
...
         position += 2
     end
     return vertices
 end
 
-function decompose{FT <: Face, T}(::Type{FT}, c::Cylinder{2, T}, resolution = (2, 2))
-    r = SimpleRectangle{T}(c.origin[1] - c.r/2, c.origin[2], c.r, height(c))
-    return decompose(FT, r, resolution)
+function decompose{FT<:Face,T}(::Type{FT},c::Cylinder{2,T},resolution=(2,2))
+    r = SimpleRectangle{T}(c.origin[1]-c.r/2,c.origin[2],c.r,height(c))
```
Comment (SimonDanisch) [inline src/decompose.jl]: You reverted my changes as well: You incorrectly return always `Face{3, Int}` and ignore the requested face type.

**Comment by SimonDanisch** [comment]
Thanks, this looks more efficient! But you completely reverted all my style and correctness changes ....

**Comment by edljk** [comment]
Sorry for having erased your help ! Is it better, now ?

**Comment by SimonDanisch** [comment]
Great! :)

**Comment by SimonDanisch** [comment]
Thanks a lot!

**Pull request merged**